include/nil/crypto3/zk/snark/systems/plonk/placeholder/permutation_argument.hpp

```diff
@@ -183,27 +205,41 @@ namespace nil {
                         // 2. Add commitment to V_P to transcript
 
                         // 3. Calculate h_perm, g_perm at challenge point
-                        math::polynomial_dfs<typename FieldType::value_type> g_poly;
-                        math::polynomial_dfs<typename FieldType::value_type> h_poly;
+                        math::polynomial_dfs<typename FieldType::value_type> one_polynomial(
+                            0, preprocessed_data.common_data.basic_domain->size(), FieldType::value_type::one());
+                        math::polynomial_dfs<typename FieldType::value_type> g_poly = one_polynomial;
+                        math::polynomial_dfs<typename FieldType::value_type> h_poly = one_polynomial;
 
                         for (std::size_t i = 0; i < column_polynomials_values.size(); i++) {
                             typename FieldType::value_type pp = column_polynomials_values[i] + gamma;
-                            if(i == 0){

```

Comment:
Is it feasible to use smart product, like in polynomial_product, like here: https://github.com/NilFoundation/crypto3-zk/blob/289ccd287ec43420358fd44407c92129f8966582/include/nil/crypto3/zk/snark/systems/plonk/placeholder/permutation_argument.hpp#L70

Reply:
Yes, it would probably help a bit, but let's do it later.